src/app/components/register/register.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, OnInit } from '@angular/core';
+import { RegisterService } from '../../services/register/register.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  username: string;
+  password: string;
+  confirmPsw: string;
+  usernameError: string;
+  passwordError: string;
+  confirmPswError: string;
+  messageSuccess: string;
+  registerStatus: boolean;
+
+  constructor(private registerService: RegisterService, private router: Router) { }
+
+  ngOnInit() { }
+
+  CheckUserName(username: string): boolean {
+    if (username !== undefined && username.length > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  myValidateFunction(username: string) {
+    const pattern = /^[A-Za-z0-9]\w{0,}$/;
+    if (pattern.test(username)) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
return (pattern.test(username));

Reply:
thanks for the simplification tip, my mistake